src/scripts/main.js

```diff
@@ -1,3 +1,330 @@
 'use strict';
 
 // write code here
+const table = document.querySelector('table');
+const thead = document.querySelector('thead');
+const tbody = table.querySelector('tbody');
+const rows = [...table.querySelectorAll('tbody tr')];
```

Comment:
The `rows` array is initialized once and reused for sorting. After adding new rows, this array will not include them, leading to potential sorting issues. Consider updating the `rows` array inside each sorting function to include all current rows.